sdk/identity/Azure.Identity/src/RegionalAuthority.cs

```diff
@@ -14,46 +13,7 @@ namespace Azure.Identity
     /// </summary>
     internal readonly struct RegionalAuthority : IEquatable<RegionalAuthority>
```

Comment:
Should we just remove this class altogether and use `AzureLocation` instead?

cc @schaabs

Reply:
There are a few clouds here that AzureLocation doesn't have.  If those are simply "missed" in AzureLocation then that might be an option, but if this class needs to have private clouds that we don't necessarily want to expose in AzureLocation then having this can normalize it.

Reply:
Another option would be to add the locations that identity might need that are unique as internal static AzureLocation instances.  We could then delete this class and update the internal references to use AzureLocation directly.

Reply:
Should have region name definitions for UsGov/China clouds defined in core regardless? Either new regions+dictionaries alongside [PublicCloudLocations](https://github.com/Azure/azure-sdk-for-net/blob/d08c3c2da09fbfe7611300380dcecfd21be65925/sdk/core/Azure.Core/src/AzureLocation.cs#L18) or as separate classes?

Reply:
@christothes is correct here, we should use `AzureLocation` from Azure.Core rather than having our own extensible enum. We should be in sync with the values in `AzureLocation` and if there are discrepancies it's most likely because I took the values for this from the `Region` class in track 1 which is possibly out of date. Either way `RegionalAuthority` was really just meant to be a list of valid Azure regions / locations. I only appended `Authority` to the type name so if we had to GA before it was added to core they would not conflict, but since we haven't GA'd this type yet we can simply remove it and reference the type in core instead.

Reply:
Ok, these are the list of items not in azure.core

- internal const string NorwayEastValue = "norwayeast";
- internal const string ChinaNorthValue = "chinanorth";
- internal const string ChinaEastValue = "chinaeast";
- internal const string ChinaNorth2Value = "chinanorth2";
- internal const string ChinaEast2Value = "chinaeast2";
- internal const string GermanyCentralValue = "germanycentral";
- internal const string GermanyNorthEastValue = "germanynortheast";
- internal const string GovernmentUSVirginiaValue = "usgovvirginia";
- internal const string GovernmentUSIowaValue = "usgoviowa";
- internal const string GovernmentUSArizonaValue = "usgovarizona";
- internal const string GovernmentUSTexasValue = "usgovtexas";
- internal const string GovernmentUSDodEastValue = "usdodeast";
- internal const string GovernmentUSDodCentralValue = "usdodcentral";

We will have to do a patch release in azure.core to add the values that are missing.  Do we know if all of these are needed / used?  My assumption is yes but just want to verify since they will be public in azure.core vs internal where they are now.  If they all are I will create a PR to add these to azure.core and work with @AlexGhiondea on releasing a 1.22.1 or wait for 1.23.

Gonna mark this as draft for now until that happens so no one merges it.